### ISSUE TITLE
Parsing a method with "//" inline comments

### ISSUE THREAD

**Issue by vignesh0710** [javaparser/javaparser#2561]
Trying to parse 

1. A function with ```// some sample comment```  in the function
2. Get method invocations in the function. 

Since the method has ```//``` I use the following. approach

```

import com.github.javaparser.ast.body.*;
import com.github.javaparser.*;
import java.io.*;
import java.util.*;

import com.github.javaparser.ast.comments.BlockComment;
import com.github.javaparser.ast.expr.MethodCallExpr;
import java.nio.file.*;
import java.nio.charset.StandardCharsets;
import com.github.javaparser.ast.stmt.BlockStmt;

//Parsing the function
String snippet = "//this is a sample comment public void testGetLineEnd() //getting inputs {Layout layout = new MockLayout(LAYOUT_TEXT, mTextPaint, mWidth, mAlign, mSpacingMult, mSpacingAdd); assertEquals(2, layout.getLineEnd(1)); }";
String block = "{"+snippet+"}";
BlockStmt bs = StaticJavaParser.parseBlock(block);

//getting method invocations within the function
List<MethodCallExpr> mces = new ArrayList<>();
bs.walk(MethodCallExpr.class, mce-> mces.add(mce) );
```

```
Error:

Exception in thread "main" com.github.javaparser.ParseProblemException: (line 1,col 1) Parse error. Found <EOF>, expected "}"
Problem stacktrace : 
  com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:10667)
  com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:10526)
  com.github.javaparser.GeneratedJavaParser.Block(GeneratedJavaParser.java:3872)
  com.github.javaparser.GeneratedJavaParser.BlockParseStart(GeneratedJavaParser.java:5748)
  com.github.javaparser.JavaParser.parse(JavaParser.java:123)
  com.github.javaparser.JavaParser.parseBlock(JavaParser.java:317)
  com.github.javaparser.StaticJavaParser.parseBlock(StaticJavaParser.java:241)
  com.example.helloworld.TestClass.main(TestClass.java:32)

	at com.github.javaparser.StaticJavaParser.handleResult(StaticJavaParser.java:260)
	at com.github.javaparser.StaticJavaParser.parseBlock(StaticJavaParser.java:241)
	at com.example.helloworld.TestClass.main(TestClass.java:32)

Process finished with exit code 1

```

Any suggestions on how to fix this would be great!

**Comment by matozoid** [comment]
Try printing the content of block...

**Comment by vignesh0710** [comment]
I tried and got the following OP:

```
{//this is a sample comment public void testGetLineEnd()  {Layout layout = new MockLayout(LAYOUT_TEXT, mTextPaint, mWidth, mAlign, mSpacingMult, mSpacingAdd); assertEquals(2, layout.getLineEnd(1)); }}

```

It looks correct though, I am adding the open close braces to the snippet since it has comments with ```//``` and the parser doesnt know the start and end of the function. 
Still not sure what is causing the error.

**Comment by matozoid** [comment]
Try putting `class Test {` and `}` around it, and compiling it.

**Comment by matozoid** [comment]
Don't forget to check out the Javadoc, by the way: https://www.javadoc.io/static/com.github.javaparser/javaparser-core/3.15.15/com/github/javaparser/ast/body/BodyDeclaration.html

**Comment by vignesh0710** [comment]
Getting a different error.
Also its a function and the end goal is to parse and get some outputs out of it. I took this approach of adding ```{}``` to the snippet from a previous question and it used to work really confused on what's the. issue now. Can you please help?

**Comment by matozoid** [comment]
These are basic Java questions and not about JavaParser, so this is not the right place to ask these questions. I'll ask on the Gitter channel if anybody feels like teaching you Java.

**Comment by vignesh0710** [comment]
Ok. Thanks. So this error is some java syntax error is it and not JavaParser related?

**Comment by matozoid** [comment]
In general: if the compiler can't compile it, JavaParser can't parse it.

**Comment by edefazio** [comment]
Ok, I'll give this a crack...
I think your assumptions about what Javaparser does may be the problem here...

Yes, you can pass in a String, but the JavaParser has no idea where to put linebreaks, so when you 
copy the text (inside the " "'s) into your Java IDE (like eclipse or IntelliJ)... it doesn't compile... because when you put a line comment `//` in front of the code on the same line, the Java compiler (and JavaParser) correctly assumes EVERYTHING after the `//` on the same line is a textual comment, and therefore NOT code in Javas AST.

if you just change your code a bit (and add a line breaks)
```java
        String snippet =
            "//this is a sample comment \n" +
            "public void testGetLineEnd(){\n" +
            "//getting inputs \n" +
            "Layout layout = new MockLayout(LAYOUT_TEXT, mTextPaint, mWidth, mAlign, mSpacingMult, mSpacingAdd); assertEquals(2, layout.getLineEnd(1));\n" +
            "}";
        BodyDeclaration bd = StaticJavaParser.parseBodyDeclaration(snippet);
```
...also what you are parsing (in the above case) is what is called a `BodyDeclaration` here specifically a `MethodDeclaration` in JavaParser land (its NOT a `BlockStmt`, a `BlockStmt` is the statements within the `{ }`s)

Any code youd like to parse should probably be looked at in a Java IDE first to make sure it's valid Java... and make sure if you Stringify it like above, you add line breaks... then make sure you understand the terminology for the components of JavaParser... USUALLY its easiest to just build a wrapper (class as a String) like @matozoid explains above that contains whatever you are trying to parse, then build the AST by `StaticJavaParser.parse( aStringRepresentingAClass )` I think if you follow these steps you'll be on the right track.

In short: the assumption you can copy paste a snippet of code (that is a snippet and not valid Java code from the Java compilers perspective) into JavaParser and let it figure out how to "fix it"(i.e. where the line breaks go & what "part" of code the String represents) is not correct.

**Comment by MysterAitch** [comment]
Hi @vignesh0710, it looks like the question has been answered and it has been over a month since the last reply, so I'm going to close this issue to help keep the issue tracker tidy.

Please do reply/follow up if there is anything outstanding!

Stay safe and take care :)
